#Titanic data exploration

Load the necessary python modules to explore the data.

In [6]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

Load the titanic data using pandas

In [8]:
train = pd.read_csv('data/train.csv', index_col='PassengerId')

Let's start by describing the data and by getting a feel for any missing observations.

In [13]:
train.describe()

Survived      Pclass         Age       SibSp       Parch        Fare
count  891.000000  891.000000  714.000000  891.000000  891.000000  891.000000
mean     0.383838    2.308642   29.699118    0.523008    0.381594   32.204208
std      0.486592    0.836071   14.526497    1.102743    0.806057   49.693429
min      0.000000    1.000000    0.420000    0.000000    0.000000    0.000000
25%      0.000000    2.000000   20.125000    0.000000    0.000000    7.910400
50%      0.000000    3.000000   28.000000    0.000000    0.000000   14.454200
75%      1.000000    3.000000   38.000000    1.000000    0.000000   31.000000
max      1.000000    3.000000   80.000000    8.000000    6.000000  512.329200

In [14]:
train.isnull().sum()

Survived      0
Pclass        0
Name          0
Sex           0
Age         177
SibSp         0
Parch         0
Ticket        0
Fare          0
Cabin       687
Embarked      2
dtype: int64

Whoa. What is up with Cabin and Age? If we want to use those, we're going to have to figure something out to do with all of those missing values. But we will look into that later. Let's look at how the various features affect whether a passenger survived. But first, let us create a binary variable for each passenger's sex - '0' for male, '1' for female.

In [18]:
train['Sx'] = np.where(train.Sex=='male',0,1)

In [17]:
train.groupby('Survived').describe()

Age        Fare       Parch      Pclass       SibSp  \
Survived                                                                     
0        count  424.000000  549.000000  549.000000  549.000000  549.000000   
         mean    30.626179   22.117887    0.329690    2.531876    0.553734   
         std     14.172110   31.388207    0.823166    0.735805    1.288399   
         min      1.000000    0.000000    0.000000    1.000000    0.000000   
         25%     21.000000    7.854200    0.000000    2.000000    0.000000   
         50%     28.000000   10.500000    0.000000    3.000000    0.000000   
         75%     39.000000   26.000000    0.000000    3.000000    1.000000   
         max     74.000000  263.000000    6.000000    3.000000    8.000000   
1        count  290.000000  342.000000  342.000000  342.000000  342.000000   
         mean    28.343690   48.395408    0.464912    1.950292    0.473684   
         std     14.950952   66.596998    0.771712    0.863321    0.708688   
         min      0.420000    0.000000    0.000000    1.000000    0.000000   
         25%     19.000000   12.475000    0.000000    1.000000    0.000000   
         50%     28.000000   26.000000    0.000000    2.000000    0.000000   
         75%     36.000000   57.000000    1.000000    3.000000    1.000000   
         max     80.000000  512.329200    5.000000    3.000000    4.000000   

                        Sx  
Survived                    
0        count  549.000000  
         mean     0.147541  
         std      0.354968  
         min      0.000000  
         25%      0.000000  
         50%      0.000000  
         75%      0.000000  
         max      1.000000  
1        count  342.000000  
         mean     0.681287  
         std      0.466660  
         min      0.000000  
         25%      0.000000  
         50%      1.000000  
         75%      1.000000  
         max      1.000000

68% of the survivors are female. Also, it appears that the Survivors were more likely to be part of first or second class, and paid more for their tickets